uh50_api/uh50.py

```diff
@@ -2,70 +2,300 @@
+        match = re.search(UH50_REGEX_CONFIG['yearly_set_day'], str(data), re.M|re.I)
+        if match:
+            self.yearly_set_day = match.group(1)
+
+        # monthly_set_day
+        match = re.search(UH50_REGEX_CONFIG['monthly_set_day'], str(data), re.M|re.I)
+        if match:
+            self.monthly_set_day = match.group(1)
+
+        # meter_date_time
+        match = re.search(UH50_REGEX_CONFIG['meter_date_time'], str(data), re.M|re.I)
+        if match:
+            try:
+                self.meter_date_time = datetime.strptime(match.group(1),'%Y-%m-%d&%H:%M:%S')
+            except:
+                self.meter_date_time = match.group(1)
+
+        # measuring_range_m3ph
+        match = re.search(UH50_REGEX_CONFIG['measuring_range_m3ph'], str(data), re.M|re.I)
+        if match:
+            try:
+                self.measuring_range_m3ph = float(match.group(1))
+            except:
+                self.measuring_range_m3ph = match.group(1)
+
+        # settings_and_firmware
+        match = re.search(UH50_REGEX_CONFIG['settings_and_firmware'], str(data), re.M|re.I)
+        if match:
+            self.settings_and_firmware = match.group(1)
+
+        # flow_hours
+        match = re.search(UH50_REGEX_CONFIG['flow_hours'], str(data), re.M|re.I)
+        if match:
+            try:
+                self.flow_hours = int(match.group(1))
+            except:
+                self.flow_hours = match.group(1)
+
+        return
+ 
+    def validate(self) -> Serial:
+        try:
+            result = self._connect_serial()
+            result.close
+        except:
+            raise
+
+    def _connect_serial(self) -> Serial:
+        "Make the connection to the serial device"
+        return serial.Serial(self.port,
+                            baudrate=300,
+                            bytesize=serial.SEVENBITS,
+                            parity=serial.PARITY_EVEN,
+                            stopbits=serial.STOPBITS_TWO,
+                            timeout=1,
+                            xonxoff=0,
+                            rtscts=0
+                            )
+
+    def _wake_up(self, conn) -> str:
+        "Waking up the device and get the model number"
+        # Waking up should be done at 300 baud
+        # Sending /?!
+        conn.write(b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x2F\x3F\x21\x0D\x0A")
+        ir_command='/?!\x0D\x0A'
+        conn.write(ir_command.encode('utf-8'))
+        conn.flush()
+        return conn.readline().decode('utf-8')[1:9]   # Read at 300 baud, this gives us the typenr
+
+    def _get_data(self, conn):
+        conn.baudrate=2400 # Now switch to 2400 BAUD. This could be different for other models. Let me know if you experience problems.
+        ir_lines = ""
+        ir_line = ""
+        iteration = 0
+        while ir_line != b'' and iteration < 100:
```

Comment:
Waarom worden daar 100 iteraties gedaan?

Reply:
Om een eindeloze loop te voorkomen. Wat is een betere oplossing?

Reply:
Er is hier niet perse een betere oplossing. De vraag is meer: waarom 100 en niet 69 of 42.

Reply:
Ah zo. Ja dat is 27 eigenlijk wel voldoende, langer wordt het niet.

Reply:
ok, maar als je hier 27 neerzet, blijft bij mij als lezer dezelfde vraag bestaan. Wat betekent 27? Het is gebruikelijk om voor dit soort magic numbers een constante aan te maken met een naam die beschrijft wat het betekent. Als ik je goed begrijp, komen er max 27 regels terug?! Een constante met de naam MAX_NUMBER_OF_LINES zou dan de vraag meteen beantwoorden. Nu ik dit zo typ; is dit iets wat afwijkt per heatmeter? Dan is het geen constante, maar config.

Reply:
Ja dat klinkt goed. Het is overigens (vermoedelijk) constant, maar ik heb hier maar 1 apparaat om het te testen.